### PR TITLE
MPDX-6920: Fix Storybook Renders

### PR DESCRIPTION
**Pull request by reldredge71**
Fixed the Storybook tests that were producing errors.  Also added error messages to the components and wrote tests for them.

### CODE REVIEW

File: src/components/Task/Drawer/CommentList/CommentList.tsx [hunk 1]
```diff
@@ -53,7 +53,9 @@ const TaskDrawerCommentList = ({
   return (
     <>
       <Box m={2}>
-        {loading ? (
+        {error ? (
+          <p>{error.toString()}</p>
```
Comment (reldredge71) [inline src/components/Task/Drawer/CommentList/CommentList.tsx:57]: This is very bare bones, not sure how we want to render error messages

Comment (OzzieOrca) [reply src/components/Task/Drawer/CommentList/CommentList.tsx:57]: Christian was using the snackbar. https://github.com/CruGlobal/mpdx-react/pull/40/files#diff-c2a85f1c4f401282f7b2bffdfc7e006980106f80113c4e0668b8d798ea616643R260

Seems ok. We haven't discussed any best practices but I assume Tatai was the one who brought that in. Not sure how that would work during render... Maybe it needs a useEffect (or a custom hook to watch it and call the snackbar). Or we could make an error icon component with a message that could be rendered in the body of the list.

**Review by OzzieOrca — APPROVED** [review]
Thanks :) Looks good.

File: src/components/Task/Drawer/CommentList/CommentList.tsx [hunk 2]
```diff
@@ -44,6 +46,12 @@ const TaskDrawerCommentList = ({
     },
   });
 
+  useEffect(() => {
```
Comment (OzzieOrca) [inline src/components/Task/Drawer/CommentList/CommentList.tsx:49]: I may have misled you with the `useEffect` suggestion. It's fine but do you think `onError` https://www.apollographql.com/docs/react/api/react/hooks/#options would be better? Idk how easy that is to test. Hopefully it's easy.

File: src/components/Task/Drawer/CommentList/CommentList.tsx [hunk 3]
```diff
@@ -76,8 +82,8 @@ const TaskDrawerCommentList = ({
                 </CardContent>
               </Card>
             )}
-            {data.task.comments.nodes.length > 0 &&
-              data.task.comments.nodes.reduce((result, comment) => {
+            {data?.task.comments.nodes?.length > 0 &&
+              data.task.comments.nodes?.reduce((result, comment) => {
```
Comment (OzzieOrca) [inline src/components/Task/Drawer/CommentList/CommentList.tsx:86]: This is defined if the last condition was true
```suggestion
              data.task.comments.nodes.reduce((result, comment) => {
```

File: src/components/Task/Drawer/CommentList/CommentList.tsx [hunk 4]
```diff
@@ -74,8 +82,8 @@ const TaskDrawerCommentList = ({
...
```
Comment (OzzieOrca) [inline src/components/Task/Drawer/CommentList/CommentList.tsx:86]:
```suggestion
              data.task.comments.nodes.reduce((result, comment, index, nodes) => {
```
See comment below

File: src/components/Task/Drawer/CommentList/CommentList.tsx [hunk 5]
```diff
@@ -85,7 +93,7 @@ const TaskDrawerCommentList = ({
                     <TaskDrawerCommentListItem
                       comment={comment}
                       reverse={comment.me}
-                      nextComment={data.task.comments.nodes[result.length + 1]}
+                      nextComment={data.task.comments?.nodes[result.length + 1]}
```
Comment (OzzieOrca) [inline src/components/Task/Drawer/CommentList/CommentList.tsx:96]: You didn't write this but it seems a little odd. I guess it's fine, I was just trying to think if there was a better way to do it. You could maybe make two arrays, shift the second array left one (by removing the first element), zip them together so you've got pairs of the comment and the next comment and then map but I think that's worse 😄 

Does this help make it slightly simpler? 🤷‍♂️ 
```suggestion
                      nextComment={nodes[index + 1]}
```

**Review by OzzieOrca — CHANGES_REQUESTED** [review]
Thanks, looks good. Just a couple small things.

File: src/components/Task/Drawer/CommentList/CommentList.tsx (hunk 3 above)
Comment (reldredge71) [reply src/components/Task/Drawer/CommentList/CommentList.tsx:86]: Yeah I might have been overly cautious here, but you're right.

**Review by OzzieOrca — APPROVED** [review]

**Comment by OzzieOrca** [comment]
Just dumping the Vercel logs here. Idk why the rest of the checks are hanging. Hopefully fixing the conflicts will get it going again.
Screen Shot 2021-03-25 at 2 29 25 PM

**Comment by reldredge71** [comment]
@OzzieOrca I was trying to replace all instances of `MockedProvider` with `GqlMockedProvider`.  However, I was having a lot of issues with the latter.  I couldn't tell what was going on -- the Stories kept producing errors saying something was null when I clearly defined it in the mocks.  `MockedProvider` works perfectly though.  Maybe we need to fix the implementation of `GqlMockedProvider`?

Also I would eventually like to have a way to create a Story for a loading state, without having to add "delay" to the mocks... I haven't found another way to do that yet.  Maybe we can make that a part of `GqlMockedProvider` somehow?

**Review by OzzieOrca — APPROVED** [review]

**Comment by OzzieOrca** [comment]
Sorry it's been giving you issues... We can definitely iterate on it. The implementation right now is pass everything down to Ergonomock, add the schema prop, and improve the typings of the mocks. https://github.com/CruGlobal/mpdx-react/blob/7e58576276ece2685e54c26a038334a603e5258d/__tests__/util/graphqlMocking.tsx#L31-L37 I haven't written many tests with it yet...

You have an example of a test that was giving you trouble? Theoretically it should work for both tests and stories... I haven't looked into the delays with stories yet...

**Comment by reldredge71** [comment]
@OzzieOrca I'm not sure what the exact issue was... I've had tests fail just with using `<GqlMockedProvider<QUERYTYPE>>`.  I think in my ContactsTable PR you were talking about how using a top level query made a difference.... maybe the queries I was using in this PR were not as high level as `ContactsQuery`? https://github.com/CruGlobal/mpdx-react/pull/34#discussion_r600671250

**Pull request merged**